spring-cloud-stream-binder-jms-activemq/src/main/java/org/springframework/cloud/stream/binder/jms/activemq/ActiveMQQueueProvisioner.java

```diff
@@ -65,26 +173,61 @@ private Queue createQueue(String topicName, Session session, String consumerName
 		return queue;
 	}
 
-	@Override
-	public String provisionDeadLetterQueue() {
-		Session session = null;
-		Connection connection = null;
-		try {
-			connection = connectionFactory.createConnection();
-			session = connection.createSession(true, 1);
-			session.createQueue(ACTIVE_MQ_DLQ);
-		} catch (JMSException e) {
-			e.printStackTrace();
-		}finally {
+	private final class JmsProducerDestination implements ProducerDestination {
+
+		private final Map<Integer, Topic> partitionTopics;
+
+		private JmsProducerDestination(Map<Integer, Topic> partitionTopics) {
+			this.partitionTopics = partitionTopics;
+		}
+
+		@Override
+		public String getName() {
 			try {
-				JmsUtils.commitIfNecessary(session);
-				JmsUtils.closeSession(session);
-				JmsUtils.closeConnection(connection);
-			} catch (JMSException e) {
-				e.printStackTrace();
+				return partitionTopics.get(-1).getTopicName();
 			}
+			catch (JMSException e) {
+				throw new IllegalStateException(e);
+			}
```

Comment:
M-m-m. There was my comment that we don't have here any JMS exception at all: https://github.com/spring-cloud/spring-cloud-stream-binder-jms/pull/16#discussion_r99956485